comp/core/workloadmeta/impl/store.go

```diff
@@ -407,6 +385,22 @@ func (w *workloadmeta) GetKubernetesMetadata(id string) (*wmdef.KubernetesMetada
 	return entity.(*wmdef.KubernetesMetadata), nil
 }
 
+// ListKubernetesMetadata implements Store#ListKubernetesMetadata.
+func (w *workloadmeta) ListKubernetesMetadata(filterFunc wmdef.EntityFilterFunc[*wmdef.KubernetesMetadata]) []*wmdef.KubernetesMetadata {
+	entities := w.listEntitiesByKind(wmdef.KindKubernetesMetadata)
+
+	var metadata []*wmdef.KubernetesMetadata
+	for _, entity := range entities {
+		kubeMetadata := entity.(*wmdef.KubernetesMetadata)
+
+		if filterFunc == nil || filterFunc(kubeMetadata) {
```

Comment:
Checking for nil here made me think about the implementation of the `MatchEntity` method of the filter struct [here](https://github.com/DataDog/datadog-agent/blob/d8657732e1c75441a76abd4f3d39fa3bb5686007/comp/core/workloadmeta/def/filter.go#L108-L110). 

We are not checking if the entity filter is nil. We should probably update it to avoid `nil pointer dereference panic`

Reply:
Yes, let's open a separate PR to fix that :+1: